Mathlib/MeasureTheory/Function/SimpleFunc.lean

```diff
@@ -553,6 +553,49 @@ instance instModule [Semiring K] [AddCommMonoid β] [Module K β] : Module K (α
 theorem smul_eq_map [SMul K β] (k : K) (f : α →ₛ β) : k • f = f.map (k • ·) :=
   rfl
 
+instance [NonUnitalNonAssocSemiring β] : NonUnitalNonAssocSemiring (α →ₛ β) where
+  left_distrib _ _ _ := ext fun _ ↦ left_distrib ..
+  right_distrib _ _ _ := ext fun _ ↦ right_distrib ..
+  zero_mul _ := ext fun _ ↦ zero_mul ..
+  mul_zero _ := ext fun _ ↦ mul_zero ..
+
+instance [Semiring β] : Semiring (α →ₛ β) where
+
+instance [Monoid K] [MulAction K β] : MulAction K (α →ₛ β) where
+  one_smul _ := ext fun _ ↦ one_smul ..
+  mul_smul _ _ _ := ext fun _ ↦ mul_smul ..
+
+instance [CommSemiring K] [Semiring β] [Algebra K β] : Algebra K (α →ₛ β) :=
+  RingHom.toAlgebra'
+    { toFun _ := const α <| algebraMap K β _
+      map_one' := SimpleFunc.ext fun _ => algebraMap K β |>.map_one ▸ rfl
+      map_mul' _ _:= ext fun _ => algebraMap K β |>.map_mul _ _
+      map_zero' := SimpleFunc.ext fun _ => algebraMap K β |>.map_zero ▸ rfl
+      map_add' _ _ := ext fun _ => algebraMap K β |>.map_add _ _ }
+    fun c f => ext fun _ => Algebra.commutes c <| f _
```

Comment:
You should just be using `Algebra K (α →ₛ β) where` here. As is, I'm relatively confident you're introducing a diamond for the `SMul K (α →ₛ β)` instance, because the existing instance is defined by `k • f := ⟨k • ⇑f, _, _⟩` and the one you're defining is `k • f := const α (algebraMap K β k) * f`. (You should check, and then change it regardless).

The way you check is this:
```
example : Algebra.toSMul (R := K) (A := α →ₛ β) = instSMul := by with_reducible_and_instances rfl
```
If this fails, then it's a (non-defeq) diamond.

Reply:
Oh, and you didn't include the lemma giving the definition of `algebraMap K (α →ₛ β)`.

Reply:
I guess I didn't (and perhaps still don't) understand what you meant about the lemma defining `algebraMap K (α →ₛ β)`. I thought I had just bundled this into the definition, but instead I guess I made a diamond!

Reply:
Oh...you must mean the "use lemma" for `algebraMap`...I'll write that...

Reply:
Oh wait...I guess what you are saying is if I can define an algebra map, the declaration can use it directly...that sounds good...let me try that...